crates/sui-indexer-alt-graphql/src/api/types/transaction/mod.rs

```diff
@@ -158,6 +179,159 @@ impl Transaction {
+/// tx_lo: The cp_sequence_number of the checkpoint at the start of the bounds.
+/// tx_hi: The tx_lo of the checkpoint directly after the cp_bounds.end(). If it does not exists,
+///      at cp_bounds.end() fallback to the maximum tx_sequence_number in the context's watermark
+///      (global_tx_hi).
+///
+/// NOTE: for consistency, assume that lowerbounds are inclusive and upperbounds are exclusive.
+/// Bounds that do not follow this convention will be annotated explicitly (e.g. `lo_exclusive` or
+/// `hi_inclusive`).
+async fn tx_unfiltered(
+    ctx: &Context<'_>,
+    cp_bounds: &RangeInclusive<u64>,
+    page: &Page<CTransaction>,
+    global_tx_hi: u64,
+) -> Result<Vec<u64>, RpcError> {
+    let pg_reader: &PgReader = ctx.data()?;
+    let query = query!(
+        r#"
+        WITH
+        tx_lo AS (
+            SELECT 
+                tx_lo 
+            FROM 
+                cp_sequence_numbers 
+            WHERE 
+                cp_sequence_number = {BigInt}
+            LIMIT 1
+        ),
+
+        -- tx_hi is the tx_lo of the checkpoint directly after the cp_bounds.end()
+        tx_hi AS (
+            SELECT 
+                tx_lo AS tx_hi
+            FROM 
+                cp_sequence_numbers 
+            WHERE 
+                cp_sequence_number = {BigInt} + 1 
+            LIMIT 1
+        )
+
+        SELECT
+            (SELECT tx_lo FROM tx_lo) AS "tx_lo",
+            -- If we cannot get the tx_hi from the checkpoint directly after the cp_bounds.end() we use global tx_hi.
+            COALESCE((SELECT tx_hi FROM tx_hi), {BigInt}) AS "tx_hi";"#,
+        *cp_bounds.start() as i64,
+        *cp_bounds.end() as i64,
+        global_tx_hi as i64
+    );
+
+    let mut conn = pg_reader
+        .connect()
+        .await
+        .context("Failed to connect to database")?;
+
+    let results: Vec<TxBounds> = conn
+        .results(query)
+        .await
+        .context("Failed to execute query")?;
+
+    let (tx_lo, tx_hi) = results
+        .first()
+        .ok_or_else(|| RpcError::from(anyhow::anyhow!("No valid checkpoints found.")))
+        .map(|bounds| (bounds.tx_lo as u64, bounds.tx_hi as u64))?;
+
+    // Inclusive cursor bounds
+    let pg_lo = page.after().map_or(tx_lo, |cursor| cursor.max(tx_lo));
+    let pg_hi = page
+        .before()
+        .map(|cursor| cursor.saturating_add(1))
+        .map_or(tx_hi, |cursor| cursor.min(tx_hi));
+
+    const PAGINATION_OVERHEAD: usize = 2; // For has_previous_page and has_next_page calculations.
+
+    Ok(if page.is_from_front() {
+        (pg_lo..pg_hi)
+            .take(page.limit() + PAGINATION_OVERHEAD)
+            .collect()
+    } else {
+        // Graphql last syntax expects results to be in ascending order. If we are paginating backwards,
+        // we reverse the results after applying limits.
+        let mut results = (pg_lo..pg_hi)
+            .rev()
+            .take(page.limit() + PAGINATION_OVERHEAD)
+            .collect::<Vec<_>>();
```

Comment:
```suggestion
        let mut results: Vec<_> = (pg_lo..pg_hi)
            .rev()
            .take(page.limit() + PAGINATION_OVERHEAD)
            .collect();
```

Although would the following be possible?

```rust
(pg_lo..pg_hi)
    .rev()
    .take(page.limit() + PAGINATION_OVERHEAD)
    .rev()
    .collect()
```